### PR TITLE
Hint users to install development package for missing `pyconfig.h`

### PR DESCRIPTION
**Pull request by mishaturnbull**
* On missing `pyconfig.h` file, prompt to install `python-dev`
* Add `.spyproject`to `.gitignore`

See https://github.com/pyinstaller/pyinstaller/issues/1539 for discussion that prompted this pull request.

### CODE REVIEW

File: PyInstaller/building/utils.py [hunk 1]
```diff
@@ -492,6 +492,20 @@ def format_binaries_and_datas(binaries_or_datas, workingdir=None):
...
+            # installing python-dev
+            if 'pyconfig.h' in src_root_path_or_glob:
+                msg = """Unable to find "%s" when adding binary and data files.
+This would mean your Python installation doesn't come with proper library files.
+This usually happens by missing development package, or unsuitable build parameters of Python installation.
+* On Debian/Ubuntu, you would need to install Python development packages
+  * apt-get install python3-dev
+  * apt-get install python-dev
+* If you're building Python by yourself, please rebuild your Python with `--enable-shared` (or, `--enable-framework` on Darwin)
+""" % src_root_path_or_glob
```
Comment (htgoebel) [inline PyInstaller/building/utils.py]: Having two `raise` statements side-by side is a bit hard to read IMHO. Also the text of the message is duplicate. How about this:
```
if not src_root_paths:
    msg = 'Unable to find "%s" when adding binary and data files.' % (
            src_root_path_or_glob))
    if src_root_path_or_glob.endswith(…)
        msg += """\n…"""
    raise SystemExit(msg)
```
 What do you think?

File: PyInstaller/building/utils.py [hunk 2]
```diff
@@ -492,6 +492,20 @@ def format_binaries_and_datas(binaries_or_datas, workingdir=None):
             src_root_paths = glob.glob(src_root_path_or_glob)
 
         if not src_root_paths:
+            
+            # on Debian/Ubuntu, a missing pyconfig.h can be fixes with
+            # installing python-dev
+            if 'pyconfig.h' in src_root_path_or_glob:
```
[the remark this answers is not in the document]
Comment (mishaturnbull) [reply PyInstaller/building/utils.py]: Agreed -- I hadn't thought of that.

File: PyInstaller/building/utils.py (hunk 1 above)
Comment (mishaturnbull) [reply PyInstaller/building/utils.py]: I agree.  Should I do the `raise` with an empty line before it?  I think that might be more readable:

```python
if not src_root_paths:
    msg = 'Unable to find "%s" when adding binary and data files.' % (
                  src_root_path_or_glob))
    if src_root_path_or_glob.endswith("pyconfig.h"):
        msg += """blah
...
blah"""
        
    raise SystemExit(msg)

Comment (htgoebel) [reply PyInstaller/building/utils.py]: If any I'd prefer an empty line behind. But both may fail the linter :-}

Comment (mishaturnbull) [reply PyInstaller/building/utils.py]: I'll pray to the lint gods that it passes!

**Comment by mishaturnbull** [comment]
I made the changes requested to the code.  I did my best at splitting the commit in two, but it seems that I now just have three commits and a merge that all do the same thing... I really need to get a better understanding of Git.

**Comment by htgoebel** [comment]
Thanks for trying :-) git is a powerful tool but with power comes complexity.

Wasn't https://pyinstaller.readthedocs.io/en/latest/development/pull-requests.html#updating-a-pull-request helpful? What is missing there?

**Comment by mishaturnbull** [comment]
I followed that guide, but whenever I tried to push I got a "detached HEAD" error message.  I think it was caused by the fact that I had already pushed changes to origin.

**Comment by htgoebel** [comment]
So something is missing (or unclear) in that description. I assume you've been in "detached" mode earlier already. Did you checkout on a commit-id instead of a branch-name?

**Comment by mishaturnbull** [comment]
I may well have done that... should I try it again?

**Comment by htgoebel** [comment]
Yes please :-)

**Comment by mishaturnbull** [comment]
I think I got it!  Two commits, showing on my branch as the only two since the fork point.  Hope this looks good :)

**Comment by mishaturnbull** [comment]
Any ideas why the CI checks are failing?  The TravisCI hasn't yet built successfully on any of my commits, but the AppVeyor was working yesterday... seems to be something about a missing `nspkg`.  Is this my fault?

**Comment by htgoebel** [comment]
> I think I got it! Two commits, showing on my branch as the only two since the fork point. Hope this looks good :)

:-)

Ci tests partially failed due to issues with OSX.I just closes and reopend the pull-request to have test running merged with current HEAD.

**Comment by mishaturnbull** [comment]
Sounds good -- looks like I failed the TravisCI linters for a few minor issues, which I've fixed and committed.  The other failure in Travis was in the nightly build, regarding CDLL's, which I don't believe has to do with my changes.  If I'm wrong, let me know...

**Comment by htgoebel** [comment]
Thanks for the pull-request.
Merged manually after updating the commit-messages.

**Comment by mishaturnbull** [comment]
Awesome!  Thanks for the help!

**Pull request closed without merging**